tests/test_metadata.py

```diff
@@ -937,6 +937,93 @@ def test_metadata_from_stac_temporal_dimension(tmp_path, stac_dict, expected):
         assert not metadata.has_temporal_dimension()
 
 
+@pytest.mark.skipif(
+    not _PYSTAC_1_9_EXTENSION_INTERFACE,
+    reason="No backport of implementation/test below PySTAC 1.9 extension interface",
+)
+@pytest.mark.parametrize(
+    ["stac_dict", "expected"],
+    [
+        (
+            # Item without cube:dimensions metadata -> assume spatial dimensions x and y
+            StacDummyBuilder.item(),
+            {"x": [None, None], "y": [None, None]},
```

Comment:
if no dimension metadata is present, I would assume a default cube with also temporal and bands dimension

Reply:
yes indeed, that is the topic of https://github.com/Open-EO/openeo-python-client/issues/743

this PR is just about improving the spatial extent detection if there is STAC metadata about that